### PR TITLE
Add dedicated allocations page

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a counterpart to the "Observing Runs" page, but for the robotic allocations.

### CODE REVIEW

**Review by guynir42 — COMMENTED** [review]
This looks a lot like #2548 so I would apply all the comments on that PR to this one, and then ping me for another look.

File: config.yaml.defaults [hunk 1]
```diff
@@ -74,6 +74,8 @@ app:
       component: Filter
     - path: "/runs"
       component: ObservingRunPage
+    - path: "/allocations"
+      component: AllocationPage
```
Comment (guynir42) [inline config.yaml.defaults:78]: Probably want to put this below GCN events so it lines up with instruments and telescopes.

File: config.yaml.defaults [hunk 2]
```diff
@@ -119,6 +121,10 @@ app:
       icon: LocalCafe
       url: /runs
 
+    - name: Allocations
+      icon: Satellite
```
Comment (guynir42) [inline config.yaml.defaults:125]: Maybe some clock icon? there must be a bunch of time-related icons that better fit this.

File: static/js/components/NewAllocation.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        ),
+        default: -1,
+        title: "Group",
+      },
+      _altdata: {
+        type: "string",
+        title: "Alternative json data (i.e. {'slack_token': 'testtoken'}",
+      },
+    },
+    required: ["pi", "start_date", "end_date", "instrument_id"],
```
Comment (guynir42) [inline static/js/components/NewAllocation.jsx:72]: hours allocated should be required? I tried posting without it and it didn't work...

File: static/js/components/NewAllocation.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      start_date: {
+        type: "string",
+        format: "datetime",
+        title: "Start Date",
+      },
+      end_date: {
+        type: "string",
+        format: "datetime",
+        title: "End Date",
+      },
```
Comment (guynir42) [inline static/js/components/NewAllocation.jsx:41]: Is it really important to have the time as well as the date? If the allocations are nightly (and don't start/end at specific hours) then maybe it is easier not to have to fill out the exact times...

File: skyportal/handlers/api/allocation.py [hunk 5]
```diff
@@ -119,6 +119,7 @@ def post(self):
```
Comment (guynir42) [inline skyportal/handlers/api/allocation.py:119]: I would add a test to make sure the start date is before the end date.

**Review by guynir42 — COMMENTED** [review]
Looks good overall, some minor adjustments required.

File: static/js/components/NewAllocation.jsx (hunk 4 above)
Comment (guynir42) [reply static/js/components/NewAllocation.jsx:41]: Yeah ok it's probably best to leave it like this.

**Review by guynir42 — APPROVED** [review]
LGTM. If you merge the telescopes page you may have to pull master and fix conflicts before merging this. But other than that it looks good.

File: skyportal/handlers/api/allocation.py (hunk 5 above)
Comment (mcoughlin) [reply skyportal/handlers/api/allocation.py:119]: Added.

File: config.yaml.defaults (hunk 2 above)
Comment (mcoughlin) [reply config.yaml.defaults]: made it an hourglass.

File: static/js/components/NewAllocation.jsx (hunk 3 above)
Comment (mcoughlin) [reply static/js/components/NewAllocation.jsx]: yes good point

File: config.yaml.defaults (hunk 1 above)
Comment (mcoughlin) [reply config.yaml.defaults:78]: done

**Comment by mcoughlin** [comment]
Failure unrelated. Can you take a look and merge if happy @guynir42 ?

**Comment by guynir42** [comment]
I hope you don't mind I added a new line before the `(PI:` part. I think it now looks ok, I will merge when green.

**Comment by mcoughlin** [comment]
Your allocations page is my allocations page @guynir42

**Comment by guynir42** [comment]
I believe the problem with the test was this: https://stackoverflow.com/questions/3655549/xpath-containstext-some-string-doesnt-work-when-used-with-node-with-more

**Comment by mcoughlin** [comment]
@guynir42 Ah interesting. I am confused why it works sometimes and not others, but hopefully that does it!

**Pull request merged**